src/plugin/AbstractPlugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @abstract
+ */
+class AbstractPlugin {
```

Comment:
I am not sure if the AbstractPlugin should be in the gataway.

Reply:
Yes could be in it's own package like renew-formalism :thinking:

Reply:
Thats what I mean.